docs/eloquent-models/model-class.txt

```diff
@@ -0,0 +1,292 @@
+
+Eloquent provides the following traits to protect your data from mass
+assignment vulnerabilities:
+
+- ``$fillable`` contains the fields that are writeable in a mass assignment
+- ``$guarded`` contains the fields that are ignored in a mass assignment
+
+.. important::
+
+   We recommend using ``$fillable`` instead of ``$guarded`` to protect against
+   vulnerabilities. To learn more about this recommendation, see the
+   `Security Release: Laravel 6.18.35, 7.24.0 <https://blog.laravel.com/security-release-laravel-61835-7240>`__
+   article on the Laravel site.
+
+In the following example, the model allows mass assignment of the fields
+by using the ``$fillable`` attribute:
+
+.. literalinclude:: /includes/eloquent-models/PlanetMassAssignment.php
+   :language: php
+   :emphasize-lines: 14-19
+   :dedent:
+
+The following code example shows mass assignment of the ``Planet`` model:
+
+.. code-block:: php
+
+   $planets = [
+       [ 'name' => 'Earth', gravity => 9.8, day_length => '24 hours' ],
+       [ 'name' => 'Mars', gravity => 3.7, day_length => '25 hours' ],
+   ];
+
+   Planet::create($planets);
+
+The models saved to the database contain only the ``name`` and ``gravity``
+fields since ``day_length`` is omitted from the ``$fillable`` attribute.
+
+To learn how to change the behavior when attempting to fill a field omitted
+from the ``$fillable`` array, see `Mass Assignment Exceptions <https://laravel.com/docs/{+laravel-docs-version+}/eloquent#mass-assignment-exceptions>`__
+in the Laravel docs.
+
+.. _laravel-model-pruning:
+
+Specify Pruning Behavior
+------------------------
+
+Eloquent lets you specify criteria to periodically delete model data that you
+no longer need.  When you schedule or run the ``model:prune`` command,
+Laravel calls the ``prunable()`` method on all models that import the
+``Prunable`` and ``MassPrunable`` traits to match the models for deletion.
+
+To use this feature with models that use MongoDB as a database, add the
+appropriate import to your model:
+
+- ``MongoDB\Laravel\Eloquent\Prunable`` optionally performs a cleanup
+  step before deleting a model that matches the criteria
+- ``MongoDB\Laravel\Eloquent\MassPrunable`` deletes models that match the
+  criteria without fetching the model data
+
+To learn more about the pruning feature, see `Pruning Models <https://laravel.com/docs/{+laravel-docs-version+}/eloquent#pruning-models>`__
+in the Laravel docs.
+
+Prunable Example
+~~~~~~~~~~~~~~~~
+
+The following example model shows how to perform a prune that calls the
+``pruning()`` method before deleting a model that matches the query in the
+``prunable()`` method:
+
+.. literalinclude:: /includes/eloquent-models/PlanetPrune.php
+   :language: php
+   :dedent:
+
+Mass Prunable Example
+~~~~~~~~~~~~~~~~~~~~~
+
+The following example model shows how to perform a "mass prune" which deletes
+all models that match the query in the ``prunable()`` method:
```

Comment:
The most important for this feature it that you must use `MongoDB\Laravel\Eloquent\MassPrunable` instead of the one provided by Laravel. So that it supports the MongoDB's `MongoDB\Laravel\Eloquent\SoftDeletes`.

Otherwise we are only duplicating Laravel's doc.

Reply:
In the enclosing section, I mentioned the MongoDB packages to clarify that those packages are "for models that use MongoDB as a database", but will repeat the info in the example sections to make sure.

Reply:
Illuminate's MassPrunable will not work with MongoDB's SoftDelete.